### ISSUE TITLE
Do I need to limit GC Heap size when deploying to Kubernetes with windows containers?

### ISSUE THREAD

**Issue by e455a81e-d3ba-41a2-bc6d-7aafb1d9a5cd** [dotnet/runtime#83081]
Hello, I want to deploy a couple of ASP.NET Core applications in a Kubernetes cluster and I have some issues regarding memory usage. Kubernetes best practices suggest that I should set memory requests and limits for my pods.
Unfortunately, my pods crash quite often because they exceed my rather generous memory limits.  I have tried to use DOTNET_GCHeapHardLimit, but I am unsure if it is really necessary since no documentation regarding .NET applications on Kubernetes makes mention of this.

Is it recommended or required to limit the heap size for .NET deployments that run in Kubernetes?

Here is my setup
- .NET 6
- I am using Server GC
- Windows Containers using the `aspnet:6.0-windowsservercore-ltsc2019` base image
- The Kubernetes pod requests 300 MB of memory and is limited to 700 MB

**Comment by Maoni0** [comment]
> Unfortunately, my pods crash quite often because they exceed my rather generous memory limits.

can you please elaborate what this means? because your containers are exceeding the memory limit you set on them? without any specific GC configs, GC heap will not exceed limit * 0.75=525mb in your case. but you are not saying "I'm getting OOM from allocations on the GC heap" but rather "my container is getting killed because it exceeds the limit I set on it"? so how big is the GC heap when the pod got killed? or are you saying you want the GC heap to be under 525mb because you have other significant sources of memory usage?

**Comment by e455a81e-d3ba-41a2-bc6d-7aafb1d9a5cd** [comment]
Yes that is correct, I am getting OOM errors and sometimes Stackoverflow errors from my application. Other than some overhead by the windows base image my containers do not run anything else but my dotnet application. The information that the GC heap will not exceed 75% of the pod limit is already valueable information. This might be the reason for my problems I will test this.

**Comment by mangod9** [comment]
Hi @e455a81e-d3ba-41a2-bc6d-7aafb1d9a5cd, is it ok to close this issue?

**Comment by e455a81e-d3ba-41a2-bc6d-7aafb1d9a5cd** [comment]
Yes thanks. The information that .NET applications use up to 75% of the pod memory limit was all I needed to fix my issue.
